src/random.cr

```diff
@@ -277,6 +290,115 @@ module Random
     end
   end
 
+  # Fills a given slice with random bytes.
+  #
+  # ```
+  # slice = Bytes.new(4) # => [0, 0, 0, 0]
+  # Random.new.random_bytes(slice)
+  # slice # => [217, 118, 38, 196]
+  # ```
+  def random_bytes(buf : Bytes) : Nil
+    n = buf.size / sizeof(typeof(next_u))
+    remaining = buf.size - n * sizeof(typeof(next_u))
+
+    slice = buf.to_unsafe.as(typeof(next_u)*).to_slice(n)
+    slice.each_index { |i| slice[i] = next_u }
+
+    if remaining > 0
+      bytes = next_u
+      remaining.times do |i|
+        bits = i * 8
+        mask = typeof(next_u).new(0xff << bits)
+        buf[-i - 1] = UInt8.new((bytes & mask) >> bits)
```

Comment:
The `-i - 1` is cryptic. It needs a note at least. 

I'd prefer to go for the simple method of creating a `base = n * sizeof(typeof(next_u))` and using `base + i` to index.